pandas/tests/groupby/test_groupby_iloc.py

```diff
@@ -0,0 +1,154 @@
+""" Test positional grouped indexing with iloc GH#42864"""
+
+import random
+
+import pandas as pd
+import pandas._testing as tm
+
+
+def test_doc_examples():
+    """Test the examples in the documentation"""
+
+    df = pd.DataFrame(
+        [["a", 1], ["a", 2], ["a", 3], ["b", 4], ["b", 5]], columns=["A", "B"]
+    )
+
+    grouped = df.groupby("A")
+    result = grouped.iloc[1:2, :]
+    expected = pd.DataFrame([["a", 2], ["b", 5]], columns=["A", "B"], index=[1, 4])
+
+    tm.assert_frame_equal(result, expected)
+
+    result = grouped.iloc[:-1, -1:]
+    expected = pd.DataFrame([1, 2, 4], columns=["B"], index=[0, 1, 3])
+
+    tm.assert_frame_equal(result, expected)
+
+
+def test_multiindex():
+    """Test the multiindex mentioned as the use-case in the documentation"""
+
+    def make_df_from_data(data):
+        rows = {}
+        for date in dates:
+            for level in data[date]:
+                rows[(date, level[0])] = {"A": level[1], "B": level[2]}
+
+        df = pd.DataFrame.from_dict(rows, orient="index")
+        df.index.names = ("Date", "Item")
+        return df
+
+    ndates = 100
+    nitems = 20
+    dates = pd.date_range("20130101", periods=ndates, freq="D")
+    items = [f"item {i}" for i in range(nitems)]
+
+    data = {}
+    for date in dates:
+        nitems_for_date = nitems - random.randint(0, 12)
+        levels = [
+            (item, random.randint(0, 10000) / 100, random.randint(0, 10000) / 100) for item in items[:nitems_for_date]
+        ]
+        levels.sort(key=lambda x: x[1])
+        data[date] = levels
+
+    df = make_df_from_data(data)
+    result = df.groupby("Date").iloc[3:-3]
+
+    sliced = {date: data[date][3:-3] for date in dates}
+    expected = make_df_from_data(sliced)
+
+    tm.assert_frame_equal(result, expected)
+
+
+def test_against_head_and_tail():
+    """Test gives the same results as grouped head and tail"""
+
+    n_groups = 100
+    n_rows_per_group = 30
+
+    data = {
+        "group": [
+            f"group {g}" for j in range(n_rows_per_group) for g in range(n_groups)
+        ],
+        "value": [
+            random.randint(0, 10000) / 100
+            for j in range(n_rows_per_group)
+            for g in range(n_groups)
+        ],
+    }
+    df = pd.DataFrame(data)
+    grouped = df.groupby("group")
+
+    for i in [1, 5, 29, 30, 31, 1000]:
+        result = grouped.iloc[:i, :]
+        expected = grouped.head(i)
```

Comment:
its ok to test against .head() (as well) but use an expected dataframe that is manually constrcructed